### PR TITLE
(LTH-63) Fix dynamic linking in Leatherman

### PR DESCRIPTION
**Pull request by MikaelSmith**

### CODE REVIEW

File: curl/CMakeLists.txt [hunk 1]
```diff
@@ -28,6 +28,10 @@ if (BUILDING_LEATHERMAN)
 endif()
 
 add_leatherman_library(src/client.cc src/request.cc src/response.cc)
+if (WIN32)
+    symbol_exports(leatherman_curl "${CMAKE_CURRENT_LIST_DIR}/inc/leatherman/curl/export.h")
```
Comment (branan) [inline curl/CMakeLists.txt]: Is this something that should be folded into the `add_leatherman_library` macro? Something like `add_leatherman_library(src1 src2 src3 EXPORTS /path/to/export.h`?

Comment (MikaelSmith) [inline curl/CMakeLists.txt]: As an option? That could be convenient, sure. I ran into trouble doing it on other platforms though, so I should probably investigate that more first.

**Comment by MikaelSmith** [comment]
I think it's fixed up now, but going to do a few full packaging tests to check.

**Comment by MikaelSmith** [comment]
Tried el-6-x86_64 builds with standalone Leatherman (static and shared); Windows builds with static and shared libcurl and static and shared Leatherman; and an el-7-x86_64 build of PA-131.

**Comment by branan** [comment]
@MikaelSmith yeah, that's a good plan. Do you want to update the matrix as part of this PR?

**Comment by MikaelSmith** [comment]
Sure.

**Comment by MikaelSmith** [comment]
Ran into yet another Windows DLL issue :/

**Comment by branan** [comment]
Travis and appveyor changes look good. Waiting for the builds to finish

**Comment by MikaelSmith** [comment]
Passed!

**Pull request merged**